sdk/src/main/java/org/eclipse/velocitas/sdk/parser/UrlParser.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.velocitas.sdk.parser
+
+/**
+ * Provides a simplified URL parser.
+ *
+ * It is able to parse just URL starting with a "://" behind the scheme
+ * specifier, e.g. "http://somehost:1234/somePath", but not URLs like
+ * "mailto:receipient@somewhere.io".
+ * As an advantage it can handle "URLs" without leading scheme, like
+ * "//127.0.0.1:42" or "localhost:123".
+ * Currently it just provides access to the scheme and the network location
+ * ("login") part of the URL.
+ * Other elements to be added as needed ...
+ */
+class UrlParser {
```

Comment:
Is there no library in kotlin to do this?

Reply:
Well,... depends on the requirements. We can probably use [URI](https://docs.oracle.com/javase/8/docs/api/java/net/URI.htm) or even better [URL](https://docs.oracle.com/javase/8/docs/api/java/net/URL.html) classes, however these do not work with classes which don't have a leading scheme, [a feature which the UrlParser actively advertises with
](https://github.com/eclipse-velocitas/vehicle-app-cpp-sdk/blob/main/sdk/include/sdk/Utils.h#L88)

So, if we are good with always specifying URLs with `http://localhost:12345` instead of `localhost:12345` we can go for it, otherwise we will need to use the custom implementation.

Reply:
Let's stick to the current implementation to be in line with the other languages.